apps/content/services/admin/asset_recitation_audio_tracks_direct_upload_service.py

```
@@ -109,36 +97,52 @@ def finish_upload(self, key: str, upload_id: str, parts: list[dict[str, Any]]) -
+                    asset_id=asset_id,
+                    surah_number=surah_number,
+                    audio_file=key,
+                    original_filename=filename,
+                    duration_ms=duration_ms,
+                    size_bytes=size_bytes,
+                    upload_finished_at=timezone.now(),
+                )
+        except IntegrityError as exc:
+            try:
+                s3.delete_object(Bucket=settings.CLOUDFLARE_R2_BUCKET, Key=r2_key)
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)